File: TEMPLELA.cpp
#include <bits/stdc++.h>
using namespace std;

int main(int argc, char const *argv[])
{
    int t,n, num, exp_num;
    bool ans = true;
    cin>>t;
    while(t--)
    {
        cin>>n;
        exp_num = 1;
        int step = 1;
        ans = true;
        // cout<<"**********************************\n";
        for(int i=0;i<n;i++)
        {
            cin>>num;
            // cout<<"num -> "<<num << ", exp_num = " << exp_num << endl;
            if(num!=exp_num) {
                ans = false;
                // break;
            }
            if(i == n/2) {
                step = -1;
            }
            exp_num += step;
        }
        if(ans and exp_num == 0)
            cout<<"yes\n";
        else
            cout<<"no\n";
    }
    return 0;
}

File: TEMPLELA.java
import java.util.*;
import java.lang.*;
import java.io.*;

class TEMPLEA
{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while(t>0)
        {
            t -= 1;
            int n = scanner.nextInt();
            int step = 1;
            int expNUm = 1;
            int ans = 0;
            for(int i=0;i<n;i++)
            {
                int num = scanner.nextInt();
                // System.out.println("comparing " + num + " with " + expNUm);
                if(num != expNUm) {
                    ans = 1;
                }

                if(i == (n/2)){
                    step = -1;
                }
                expNUm += step;
            }

            if(ans == 0 && expNUm == 0) {
                System.out.println("yes");
            }
            else 
                System.out.println("no");
        }
    }
}